kernel/src/actions/mod.rs

```diff
@@ -152,6 +199,81 @@ impl Protocol {
             .as_ref()
             .is_some_and(|features| features.iter().any(|f| f == feature.as_ref()))
     }
+
+    /// Check if reading a table with this protocol is supported. That is: does the kernel support
+    /// the specified protocol reader version and all enabled reader features? If yes, returns unit
+    /// type, otherwise will return an error.
+    pub fn ensure_read_supported(&self) -> DeltaResult<()> {
+        match &self.reader_features {
+            // if min_reader_version = 3 and all reader features are subset of supported => OK
+            Some(reader_features) if self.min_reader_version == 3 => {
+                ensure_supported_features(reader_features, &SUPPORTED_READER_FEATURES)
+            }
+            // if min_reader_version = 3 and no reader features => ERROR
+            // note this is caught by the protocol parsing.
+            None if self.min_reader_version == 3 => unreachable!(),
+            // if min_reader_version = 1,2 and there are no reader features => OK
+            None if self.min_reader_version == 1 || self.min_reader_version == 2 => Ok(()),
+            // if min_reader_version = 1,2 and there are reader features => ERROR
+            // note this is caught by the protocol parsing.
+            Some(_) if self.min_reader_version == 1 || self.min_reader_version == 2 => {
+                unreachable!()
+            }
+            // any other min_reader_version is not supported
+            _ => Err(Error::unsupported(format!(
+                "Unsupported minimum reader version {}",
+                self.min_reader_version
+            ))),
```

Comment:
we're already converting to a string here

```suggestion
            _ => Err(Error::Unsupported(format!(
                "Unsupported minimum reader version {}",
                self.min_reader_version
            ))),
```